### PR TITLE
EG-3240-2 Updated Known Issues in CENM 1.4 release notes as per feedback [part 1]

### PR DESCRIPTION
**Pull request by ivanterziev-r3**
EG-3240-2 Updated Known Issues in CENM 1.4 release notes as per feedback [part 1]

### CODE REVIEW

File: content/en/docs/cenm/1.4/release-notes.md [hunk 1]
```diff
@@ -157,23 +157,16 @@ with multiple accounts for each task The Signing Service now prompts a specific
 * There are currently two inconsistencies in service console error codes:
   * `config-parse-error` does not display help while `config-file-not-readable` does.
   * `config-parse-error` does not colour the error code red while `config-file-not-readable` does.
-* Various CENM services handle and write new error codes to the logs inconsistently. The Signing Service, Identity Manager Service, and Network Map Service write error codes to the `DUMP` log. The Auth Service and Gateway Service write error codes to both the `OPS` and `DUMP` logs.
-* The `config-substitution-error` code is not triggered when it should be. The Angel Service for the Signing Service displays the `config-file-doesnt-exist` error code in the `DUMP` log, while the Zone Service throws an exception in the `OPS` and `DUMP` logs.
+* Some services are use logs inconsistently:
```
Comment (taylorharvey-r3) [inline content/en/docs/cenm/1.4/release-notes.md:160]:
```suggestion
* Some services are using logs inconsistently:
```

Comment (edward-prosser) [inline content/en/docs/cenm/1.4/release-notes.md:160]:
```suggestion
* Some services use logs inconsistently:
```

**Review by taylorharvey-r3 — DISMISSED** [review]
One small suggestion but LGTM

**Review by edward-prosser — APPROVED** [review]

**Review by taylorharvey-r3 — DISMISSED** [review]
LGTM

File: content/en/docs/cenm/1.4/release-notes.md [hunk 2]
```diff
@@ -157,23 +157,16 @@ with multiple accounts for each task The Signing Service now prompts a specific
 * There are currently two inconsistencies in service console error codes:
   * `config-parse-error` does not display help while `config-file-not-readable` does.
   * `config-parse-error` does not colour the error code red while `config-file-not-readable` does.
-* Various CENM services handle and write new error codes to the logs inconsistently. The Signing Service, Identity Manager Service, and Network Map Service write error codes to the `DUMP` log. The Auth Service and Gateway Service write error codes to both the `OPS` and `DUMP` logs.
-* The `config-substitution-error` code is not triggered when it should be. The Angel Service for the Signing Service displays the `config-file-doesnt-exist` error code in the `DUMP` log, while the Zone Service throws an exception in the `OPS` and `DUMP` logs.
+* Some services use logs inconsistently:
```
Comment (JayFarei) [inline content/en/docs/cenm/1.4/release-notes.md:160]: I don't think issues about logs should be expanded further. I would roll up all bullet points referring to logs in one bullet point only.
Not all services are writing error codes to the same logs - they currently can be found in `DUMP`and/or `OPS.

File: content/en/docs/cenm/1.4/release-notes.md [hunk 3]
```diff
@@ -157,23 +157,16 @@ with multiple accounts for each task The Signing Service now prompts a specific
...
-* The Admin RPC client incorrectly throws an `RpcServerException` with a `GENERAL_ERROR` code and a null message. It should throw an `RpcServerException` with a `SIGNER_SIGNABLE_MATERIAL_SOURCE_UNREACHABLE` error code.
 * On execution of `submitCertificateRevocationRequestWithLegalName` when the Identity Manager Service configuration does not have a `REVOCATION` workflow, the Admin RPC client does not throw exceptions as expected and returns empty responses instead. It should throw two `RpcServerException` exceptions with `IDENTITY_MANAGER_INVALID_REVOCATION_REASON` and `IDENTITY_MANAGER_CERTIFICATE_NOT_FOUND` codes or an exception noting that the `REVOCATION` workflow is not set up.
 *  The `config-file-doesnt-exist` error code does not appear when the Signing Service is started with a non-existing configuration file. No error code or link to relevant documentation appears in the console or logs.
```
Comment (JayFarei) [inline content/en/docs/cenm/1.4/release-notes.md:165]: I would roll up all error codes that were expected but are not in this release as: Error codes are not yet thrown consistently in logs or console across all services

File: content/en/docs/cenm/1.4/release-notes.md [hunk 4]
```diff
@@ -157,23 +157,16 @@ with multiple accounts for each task The Signing Service now prompts a specific
...
-* Network Map Service updates get stuck after the registration of about 2,500 nodes.
-* The `smr` command in the Command-line Interface (CLI) Tool is only used in CENM version 1.3. Its usage in later versions is deprecated due to the removal of the SMR (Signable Material Retriever) Service in CENM 1.4 - see the [New features and enhancements](#new-features-and-enhancements) section above for more details.
-* The Admin RPC client incorrectly throws an `RpcServerException` with a `GENERAL_ERROR` code and a null message. It should throw an `RpcServerException` with a `SIGNER_SIGNABLE_MATERIAL_SOURCE_UNREACHABLE` error code.
 * On execution of `submitCertificateRevocationRequestWithLegalName` when the Identity Manager Service configuration does not have a `REVOCATION` workflow, the Admin RPC client does not throw exceptions as expected and returns empty responses instead. It should throw two `RpcServerException` exceptions with `IDENTITY_MANAGER_INVALID_REVOCATION_REASON` and `IDENTITY_MANAGER_CERTIFICATE_NOT_FOUND` codes or an exception noting that the `REVOCATION` workflow is not set up.
```
Comment (JayFarei) [inline content/en/docs/cenm/1.4/release-notes.md:164]: I would roll this up in the generic statement

File: content/en/docs/cenm/1.4/release-notes.md [hunk 5]
```diff
@@ -157,23 +157,16 @@ with multiple accounts for each task The Signing Service now prompts a specific
...
-* There is an inconsistency in how configuration validation handles an incorrect `trustStore` location. For some parameters, a service will not start. For others, a service will try to start and will then fail.
-* The Command-line Interface (CLI) Tool should produce an error code instead of throwing Java `DUMP` exceptions (such as `java.io.FileNotFoundException`) for issues like adding the wrong file name or having a space in the path of a file.
+* When multiple CRR requests are submitted, the certificates are not updated correctly from `VALID` to `REVOKED`. This issue does not affect the CRL.
 * Information about the running version of CENM components is missing from the logs.
-* The app version is not displayed when running Helm Charts.
 * When a Signing Service is started with an incomplete or incorrect configuration, a stack trace occurs. This should be handled as an exception.
```
Comment (JayFarei) [inline content/en/docs/cenm/1.4/release-notes.md:168]: I would roll this up in the error generic statement

File: content/en/docs/cenm/1.4/release-notes.md [hunk 6]
```diff
@@ -157,23 +157,16 @@ with multiple accounts for each task The Signing Service now prompts a specific
...
-* When several CRR requests are submitted and the `plugin-ca`/`signer-ca` is used, the CRL with CRR should be signed, records in the `workflow_crr` table should be updated, and the certificate statuses should be updated from `VALID` to `REVOKED`. However, currently the `signer-ca` shows revoked node details, the records in the `workflow_crr` table are `Pending`, and certificates are `VALID` instead of `REVOKED`.
-* There is an inconsistency in how configuration validation handles an incorrect `trustStore` location. For some parameters, a service will not start. For others, a service will try to start and will then fail.
-* The Command-line Interface (CLI) Tool should produce an error code instead of throwing Java `DUMP` exceptions (such as `java.io.FileNotFoundException`) for issues like adding the wrong file name or having a space in the path of a file.
+* When multiple CRR requests are submitted, the certificates are not updated correctly from `VALID` to `REVOKED`. This issue does not affect the CRL.
 * Information about the running version of CENM components is missing from the logs.
```
Comment (JayFarei) [inline content/en/docs/cenm/1.4/release-notes.md:167]: Was this promised by our docs? if not this shouldn't be tracked as a known issue. If yes - does it precede this version? if it does this should be a known issue - if it was introduced in this version we should amend the main doc and introduce this feature when ready.

File: content/en/docs/cenm/1.4/release-notes.md [hunk 7]
```diff
@@ -157,23 +157,15 @@ with multiple accounts for each task The Signing Service now prompts a specific
 * There are currently two inconsistencies in service console error codes:
   * `config-parse-error` does not display help while `config-file-not-readable` does.
   * `config-parse-error` does not colour the error code red while `config-file-not-readable` does.
-* Various CENM services handle and write new error codes to the logs inconsistently. The Signing Service, Identity Manager Service, and Network Map Service write error codes to the `DUMP` log. The Auth Service and Gateway Service write error codes to both the `OPS` and `DUMP` logs.
-* The `config-substitution-error` code is not triggered when it should be. The Angel Service for the Signing Service displays the `config-file-doesnt-exist` error code in the `DUMP` log, while the Zone Service throws an exception in the `OPS` and `DUMP` logs.
+* Some services use logs inconsistently:
```
Comment (ivanterziev-r3) [inline content/en/docs/cenm/1.4/release-notes.md:160]:
```suggestion
* There are currently two different logs that services write error codes to (`DUMP` and `OPS`), with some services writing to both.
```

File: content/en/docs/cenm/1.4/release-notes.md [hunk 8]
```diff
@@ -157,23 +157,15 @@ with multiple accounts for each task The Signing Service now prompts a specific
...
+  * Different services write error codes to different logs - either `DUMP` only, or both `OPS` and `DUMP`.
```
Comment (ivanterziev-r3) [inline content/en/docs/cenm/1.4/release-notes.md:161]:
```suggestion
```

**Review by jmacmahonr3 — APPROVED** [review]

File: content/en/docs/cenm/1.4/release-notes.md [hunk 9]
```diff
@@ -157,23 +157,15 @@ with multiple accounts for each task The Signing Service now prompts a specific
...
+  * The Command-line Interface (CLI) Tool throws a Java DUMP exception instead of producing an error code for certain issues.
```
Comment (ivanterziev-r3) [inline content/en/docs/cenm/1.4/release-notes.md:162]:
```suggestion
```

File: content/en/docs/cenm/1.4/release-notes.md [hunk 10]
```diff
@@ -157,23 +157,13 @@ with multiple accounts for each task The Signing Service now prompts a specific
...
-* The Admin RPC client incorrectly throws an `RpcServerException` with a `GENERAL_ERROR` code and a null message. It should throw an `RpcServerException` with a `SIGNER_SIGNABLE_MATERIAL_SOURCE_UNREACHABLE` error code.
 * On execution of `submitCertificateRevocationRequestWithLegalName` when the Identity Manager Service configuration does not have a `REVOCATION` workflow, the Admin RPC client does not throw exceptions as expected and returns empty responses instead. It should throw two `RpcServerException` exceptions with `IDENTITY_MANAGER_INVALID_REVOCATION_REASON` and `IDENTITY_MANAGER_CERTIFICATE_NOT_FOUND` codes or an exception noting that the `REVOCATION` workflow is not set up.
 *  The `config-file-doesnt-exist` error code does not appear when the Signing Service is started with a non-existing configuration file. No error code or link to relevant documentation appears in the console or logs.
```
Comment (ivanterziev-r3) [inline content/en/docs/cenm/1.4/release-notes.md:163]:
```suggestion
*  Error codes are not yet thrown consistently in logs or console across all services.
```

File: content/en/docs/cenm/1.4/release-notes.md (hunk 6 above)
Comment (ivanterziev-r3) [reply content/en/docs/cenm/1.4/release-notes.md:165]: No, it was reported by QA as a problem but not really promised. Will remove.

File: content/en/docs/cenm/1.4/release-notes.md [hunk 11]
```diff
@@ -157,23 +157,13 @@ with multiple accounts for each task The Signing Service now prompts a specific
...
-* When several CRR requests are submitted and the `plugin-ca`/`signer-ca` is used, the CRL with CRR should be signed, records in the `workflow_crr` table should be updated, and the certificate statuses should be updated from `VALID` to `REVOKED`. However, currently the `signer-ca` shows revoked node details, the records in the `workflow_crr` table are `Pending`, and certificates are `VALID` instead of `REVOKED`.
-* There is an inconsistency in how configuration validation handles an incorrect `trustStore` location. For some parameters, a service will not start. For others, a service will try to start and will then fail.
-* The Command-line Interface (CLI) Tool should produce an error code instead of throwing Java `DUMP` exceptions (such as `java.io.FileNotFoundException`) for issues like adding the wrong file name or having a space in the path of a file.
+*  Error codes are not yet thrown consistently in logs or console across all services.
+* When multiple CRR requests are submitted, the certificates are not updated correctly from `VALID` to `REVOKED`. This issue does not affect the CRL.
 * Information about the running version of CENM components is missing from the logs.
```
Comment (ivanterziev-r3) [inline content/en/docs/cenm/1.4/release-notes.md:165]:
```suggestion
```

File: content/en/docs/cenm/1.4/release-notes.md (hunk 5 above)
Comment (ivanterziev-r3) [reply content/en/docs/cenm/1.4/release-notes.md:165]: Removing.

File: content/en/docs/cenm/1.4/release-notes.md [hunk 12]
```diff
@@ -157,23 +157,12 @@ with multiple accounts for each task The Signing Service now prompts a specific
...
-* There is an inconsistency in how configuration validation handles an incorrect `trustStore` location. For some parameters, a service will not start. For others, a service will try to start and will then fail.
-* The Command-line Interface (CLI) Tool should produce an error code instead of throwing Java `DUMP` exceptions (such as `java.io.FileNotFoundException`) for issues like adding the wrong file name or having a space in the path of a file.
-* Information about the running version of CENM components is missing from the logs.
-* The app version is not displayed when running Helm Charts.
+*  Error codes are not yet thrown consistently in logs or console across all services.
+* When multiple CRR requests are submitted, the certificates are not updated correctly from `VALID` to `REVOKED`. This issue does not affect the CRL.
 * When a Signing Service is started with an incomplete or incorrect configuration, a stack trace occurs. This should be handled as an exception.
```
Comment (ivanterziev-r3) [inline content/en/docs/cenm/1.4/release-notes.md:165]:
```suggestion
```

File: content/en/docs/cenm/1.4/release-notes.md (hunk 4 above)
Comment (ivanterziev-r3) [reply content/en/docs/cenm/1.4/release-notes.md:162]: Removing. Too wordy and the level of detail is not really relevant for a known issue. This is a Jira issue tracked already so we can just fix it and present it as a fixed issue in the next release (again, in a much more generic way).

File: content/en/docs/cenm/1.4/release-notes.md [hunk 13]
```diff
@@ -157,23 +157,11 @@ with multiple accounts for each task The Signing Service now prompts a specific
...
-* Network Map Service updates get stuck after the registration of about 2,500 nodes.
-* The `smr` command in the Command-line Interface (CLI) Tool is only used in CENM version 1.3. Its usage in later versions is deprecated due to the removal of the SMR (Signable Material Retriever) Service in CENM 1.4 - see the [New features and enhancements](#new-features-and-enhancements) section above for more details.
-* The Admin RPC client incorrectly throws an `RpcServerException` with a `GENERAL_ERROR` code and a null message. It should throw an `RpcServerException` with a `SIGNER_SIGNABLE_MATERIAL_SOURCE_UNREACHABLE` error code.
 * On execution of `submitCertificateRevocationRequestWithLegalName` when the Identity Manager Service configuration does not have a `REVOCATION` workflow, the Admin RPC client does not throw exceptions as expected and returns empty responses instead. It should throw two `RpcServerException` exceptions with `IDENTITY_MANAGER_INVALID_REVOCATION_REASON` and `IDENTITY_MANAGER_CERTIFICATE_NOT_FOUND` codes or an exception noting that the `REVOCATION` workflow is not set up.
```
Comment (ivanterziev-r3) [inline content/en/docs/cenm/1.4/release-notes.md:162]:
```suggestion
```

**Review by edward-prosser — APPROVED** [review]
